packages/@react-spectrum/contextualhelp/stories/ContextualHelp.stories.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {action} from '@storybook/addon-actions';
+import {Button, Flex, Link, Text} from '@adobe/react-spectrum';
+import {ContextualHelp} from '../src';
+import React from 'react';
+import {storiesOf} from '@storybook/react';
+
+storiesOf('ContextualHelp', module)
+.add(
+  'default',
+  () => <ContextualHelp title="Help title">{helpText()}</ContextualHelp>
+)
+.add(
+  'type: info',
+  () => <ContextualHelp variant="info" title="Help title">{helpText()}</ContextualHelp>
+)
+.add(
+  'with link',
+  () => (<ContextualHelp title="Help title">
+    <>
+      {helpText()}
+      <Link marginTop="size-100">Learn more</Link>
```

Comment:
To the team: the issue states that we want the link to be placed in the footer slot of a dialog, do we still want to achieve that or follow the above approach where the user needs to apply a marginTop to position the Link properly? If we wanna go ahead with the slot approach, should the api support some kind of `link` prop so we can put the link in a wrapper ourselves and give that wrapper the `footer` slot? Not sure how we can ensure that the link would automatically be placed in the `footer` slot since the link already has `link` as it's default footer slot...

Reply:
From reading the issue again. It seems like either we treat it like a dialog/popover OR we have special props. Right now, the issue is asking for special props.
```
<ContextualHelp
  title="This is the title"
  >
  children
</ContextualHelp
```
in which case, I think that if we want to have a link, then we need to also add a prop to ContextualHelp that is `link` and it'd take a React Node, `<Link wherever/>` or two props, `linkText` and `linkURL`

Otherwise, we need to just utilize Dialog's innate slots
```
<ContextualHelp>
  <Heading>This is the title</Heading>
  <Content>children</Content>
  <Footer>
     <Link wherever />
  </Footer>
</ContextualHelp
```
I'm not sure why the Content needs a specific style from ContextualHelp though. We don't clear slots though for Dialog, so theoretically, we could wrap Dialog in a Slots context and supply a classname to the content slot, and it should get merged https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/utils/src/Slots.tsx#L26

Reply:
I like option 2 personally, using the slots dialog has already defined and an API to ContextualHelp that looks similar to our other components
We can always null out other slots that Dialog exposes that we don't want to

Reply:
Here is an example of adding the `link` via a prop.

```
<ContextualHelp title="Help title" link={<Link>Learn more</Link>}>
    {helpText()}
</ContextualHelp>
```
![image](https://user-images.githubusercontent.com/1004204/146019191-f2e08f6e-ccf1-4f64-b913-96eb3a8e13c7.png)

Reply:
> I'm not sure why the Content needs a specific style from ContextualHelp though.

The custom styling shrinks down the padding used between `Dialog` slots.

Reply:
I've pushed a new proposal that uses the slots concept while hiding the detail that a `Footer` slot gets used for the link.

```
    <ContextualHelp>
      <Heading>Help title</Heading>
      <Content>Help text</Content>
      <Link>Help link</Link>
    </ContextualHelp>
```